tensorflow/contrib/timeseries/python/timeseries/head.py

```diff
@@ -71,12 +73,32 @@ def __init__(self,
     self.input_statistics_generator = input_statistics_generator
     self._name = name
 
-  def _train_ops(self, features):
-    """Add training ops to the graph."""
-    with variable_scope.variable_scope("model"):
+  @property
+  def name(self):
+    return self._name
+
+  # TODO(terrytangyuan): consolidate `model_outputs` and `_Head.LossSpec`
+  # once `_Head.create_loss` becomes extendable
+  def create_loss(self, features, mode, logits=None, labels=None):
+    """See `_Head`."""
+    with variable_scope.variable_scope(
+        "model", reuse=variable_scope.AUTO_REUSE):
```

Comment:
Is AUTO_REUSE necessary? The model should be defined from scratch in a new graph each time this is used.

Reply:
I think so - if I remove it, I will get:
```
ValueError: Variable model/state_update_step/exogenous_noise_increasing_mean/fully_connected/weights already exists, disallowed. Did you mean to set reuse=True or reuse=tf.AUTO_REUSE in VarScope
```
This is because we calculated predictions using the same variable scope before we calculate the loss:
```
with variable_scope.variable_scope("model"):
      prediction_outputs = self.model.predict(features=features)
```
Note that originally we had `reuse=True`.

Reply:
Oh good point. Can you move the variable scope out of create_loss so we can keep the explicit reuse=True/reuse=False? I'd rather have it repeated a few times but keep the "this is definitely reused/not reused" error checking.

Reply:
Sounds good. Done!